pybamm/solvers/casadi_solver.py

```diff
@@ -499,6 +515,8 @@ def create_integrator(self, model, inputs, t_eval=None, use_event_switch=False):
                     method, problem, options = self.integrator_specs[model]
                     options["grid"] = t_eval_shifted
                     integrator = casadi.integrator("F", method, problem, options)
+                    if ninputs > 1:
+                        integrator = integrator.map(ninputs, 'thread', nproc)
```

Comment:
This line should be tested. You need to solve the same model a second time with different times (i.e. an experiment) to enter this branch

Reply:
I don't think experiments work with multidimensional inputs due to the event handling - I can try to get it to work but looks complex. Might be simpler to implement parallel processing outside of the solver rather than using the map function from casadi.

Reply:
@tinosulzer this looks too hacky to implement now with the way that experiments call the solver step function. It would be worth doing though I think. Might be worth thinking about a refactor of solver, simulation, experiment.

Reply:
Oh yeah you're right it won't work with experiment. I don't see how it could work but maybe I don't know about some casadi features. Any specific reason for a refactor?

Reply:
Well the simplest thing would be to parallelize steps which means the experiment would have to control the solver but it kinda seems like it's the other way round... I am only really just looking at the solver codes in detail now but they are complicated and it might be possible to make them simpler with a refactor too but it's probably  a lot of work

Reply:
The casadi solver code is really complicated because of events

Reply:
So we can get this code running without experiments or shelve it until casadi handles events